src/entities/elastic-storage/elastic-storage-processor.js

```diff
@@ -450,15 +460,125 @@ class ElasticStorageProcessor<TxType: ByronTxType | ShelleyTxType> implements St
+    const paddedBlocks: Array<BlockData> = padEmptySlots(mappedBlocks,
+      tip.epoch, tip.slot, this.networkStartTime, this.slotsPerEpoch)
+
+    const blocksData = paddedBlocks.map((b: BlockData) => b.toPlainObject(true))
+
+    const isGenesisBlock = true
+    this.logger.debug(`storeBlocksData.constructing bulk for ${blocksData.length} slots`)
+    const blocksBody = formatBulkUploadBody(blocksData, {
+      index: this.indexFor(INDEX_SLOT),
+      getId: (o) => o.hash,
+      getData: o => ({
+        ...o,
+        ...(isGenesisBlock ? {
+          tx: undefined,
+        } : {}),
+        _chunk: chunk,
+      }),
+    })
+
+    const blockTxioToStore = blockOutputsToStore
+    this.logger.debug(`storeBlocksData.constructing bulk for ${blockTxioToStore.length} txio`)
+    const txiosBody = formatBulkUploadBody(blockTxioToStore, {
+      index: this.indexFor(INDEX_TXIO),
+      getId: (o) => o.id,
+      getData: (o) => ({
+        ...o,
+        _chunk: chunk,
+      }),
+    })
+
+    const bulkData = [...blocksBody, ...txiosBody]
+    this.logger.debug(`storeBlocksData.total bulk is ${bulkData.length} documents`)
+    const bulkChunks = _.chunk(bulkData, 1000)
+    for (let i = 0; i < bulkChunks.length; i += 1) {
+      this.logger.debug(`storeBlocksData.bulkUpload chunk ${i + 1} out of ${bulkChunks.length}`)
+      try {
+        await this.bulkUpload(bulkChunks[i])
+        await sleep(100)
+      } catch (e) {
+        this.logger.error(`Failed to bulk-upload blocks data (chunk ${i + 1} out of ${bulkChunks.length})`, e)
+        throw new Error(`Failed to bulk-upload blocks data (chunk ${i + 1} out of ${bulkChunks.length}) : ${e}`)
+      }
+    }
+
+    for (const b of paddedBlocks) {
+      // const txsData = []
+      let idx = 0
+      for (const txData of b.getTxsData()) {
+        //txsData.push(txData)
+        // collect 1k items and upload them
+        // if (idx % 1000 === 0) {
+        const txsBody = formatBulkUploadBody([txData], {
+            index: this.indexFor(INDEX_TX),
+            getId: (o) => o.hash,
+            getData: (o) => o,
+          })
+          await sleep(100)
+          await this.bulkUpload(txsBody)
+          // txsData.splice(0, txsData.length)
+        // }
+        ++idx
+        this.logger.debug('Idx:', idx)
+      }
+    }
+
+    /*
+    const txsDataGen = paddedBlocks.flatMap(b => b.getTxsData())
+    this.logger.debug(`storeBlocksData.constructing bulk for ${txsData.length} txs`)
+    const txsBody = formatBulkUploadBody(txsData, {
+      index: this.indexFor(INDEX_TX),
+      getId: (o) => o.hash,
+      getData: (o) => o,
+    })
+    */
+
+
+    // Commit every 10th chunk
+    const isCommitChunk = (chunk % 10 === 0) || isGenesisBlock
+    const sleepOnChunkMillis = (isCommitChunk ?
+      this.sleepOnCommitChunkSeconds : this.sleepOnEveryChunkSeconds) * 1000
+    if (sleepOnChunkMillis > 0) {
+      await sleep(sleepOnChunkMillis)
```

Comment:
why do we need a sleep here also? There is a 2nd sleep also for `sleepOnChunkMillis` also in this same  code block